src/pages/DatasetRegistration.js

```diff
@@ -427,16 +489,15 @@ class DatasetRegistration extends Component {
                           maxLength: '256',
                           value: this.state.datasetData.datasetName,
                           onChange: this.handleChange,
-                          className: (fp.isEmpty(this.state.datasetData.datasetName) && showValidationMessages) ?
-                            'form-control required-field-error' :
-                            'form-control',
+                          className: this.showDatasetNameErrors(this.state.datasetData.datasetName, showValidationMessages),
                           required: true,
                         }),
                         span({
                           className: 'cancel-color required-field-error-span',
-                          isRendered: fp.isEmpty(this.state.datasetData.datasetName) && showValidationMessages,
+                          isRendered: (fp.isEmpty(this.state.datasetData.datasetName) && showValidationMessages)
+                            || (!fp.isEmpty(this.state.datasetData.datasetName) && !this.validateDatasetName(this.state.datasetData.datasetName)),
                         },
-                        ['Required field']),
+                        [this.validateDatasetName(this.state.datasetData.datasetName) ? 'Required field' : 'Dataset Name already in use']),
```

Comment:
I'm not seeing this error pop up. If I create a dataset, then immediately try to create a new one with the same name, I get a 404 back from consent and a generic error message. We might need a bug fix on the POST endpoint ... have not looked closely at that yet.

![Screen Shot 2020-10-07 at 2 48 12 PM](https://user-images.githubusercontent.com/116679/95374293-27cbd900-08ac-11eb-993a-30aba392718b.png)

Reply:
Ancillary problem (and unrelated to this PR) with the above screen shot. Both buttons are disabled and the only way to close out of the modal is to click the little "X" icon.

Reply:
More info on this. On the back end, we throw a 404 in this case. I think that should really be a 409 Conflict instead. We should add a PR for that separate from this story.

Reply:
Made a ticket for the error messaging here: https://broadinstitute.atlassian.net/browse/DUOS-797

Reply:
Ticket for modal: https://broadinstitute.atlassian.net/browse/DUOS-800